BootloaderCorePkg/Library/MpInitLib/MpInitLib.c

```diff
@@ -496,7 +496,19 @@ MpInit (
     } else {
       DEBUG ((DEBUG_INFO, "MP Init (Run)\n"));
 
-      // Wait for task done
+      // The below loop waits until all APs which have started Wakeup
+      // (ApDataPtr->ApCounter) have completed wakeup
+      // (mMpDataStruct.ApDoneCounter), or the timeout is reached
+      // (AP_TASK_TIMEOUT_UNIT * AP_TASK_TIMEOUT_CNT). Whichever occurs
+      // first. This assumes that the number of running APs is never zero
+      // unless all APs have completed wakeup. This is not always the case
+      // with high core counts and tight timings, so a fixed delay here can
+      // help ensure all APs have started before this check.
+
+      if (FixedPcdGet32 (PcdCpuApInitWaitInMicroSeconds)) {
+        MicroSecondDelay (FixedPcdGet32 (PcdCpuApInitWaitInMicroSeconds));
+      }
+
```

Comment:
@bejeanmo is this delay preventing the case of 0 APs? can we also add a debug log presenting no AP is detected in such case?

Reply:
When I've seen this cause problems, it's usually that 1 or 2 APs get detected (out of 31)/